### DISCUSSION TITLE
Can I get all hyper-parameters for searching in tune_model()

### DISCUSSION THREAD

**Discussion by AlexShyXie** [pycaret/pycaret#2027] [Q&A] [answered]
Excuse me，I found the tunr_model was so quicker than gridsearch in sklearn function，I wonder which hyperparams the tune_model()  use and the detailed number for searching，is there any way to get there data？

**Comment by ngupta23** [comment, marked as answer]
you can use 

```python
models(internal=True)
```

This will return the model details including the tune_distributions (random Grid)  and tune_grid (fixed grid) used by tune_model.


Also, to get the result details from tune_model, you could do something like this

```python
tuned_model, tuner = tune_model(model, return_tuner=True)
pd.DataFrame(tuner.cv_results_) 
``` 

This will show you details about each hyper parameter that was tried.

**Comment by AlexShyXie** [comment]
> Excuse me，I found the tunr_model was so quicker than gridsearch in sklearn function，I wonder which hyperparams the tune_model()  use and the detailed number for searching，is there any way to get there data？

    Thanks very much！
